iree/compiler/Conversion/LinalgToSPIRV/SplitDispatchFunctionPass.cpp

```diff
@@ -67,33 +68,18 @@ static bool isFusableWithCurrentOpsList(
 
   linalg::LinalgOp dstOp = dyn_cast<linalg::LinalgOp>(nextOp);
   linalg::LinalgOp srcOp = dyn_cast<linalg::LinalgOp>(currOpsList.back());
-  if (dstOp && srcOp) {
-    // TODO(#2963): This splits independent linalg opreations into its own
-    // dispatch, but in reality if the iteration domain of the ops are the same,
-    // and they have all iterator types parallel, they could be put in the same
-    // dispatch region.
-    if (!dependenceGraph.hasDependenceFrom(srcOp, dstOp)) return false;
-
-#define ADD_FUSABLE_PAIR(SrcOpTy, DstOpTy, DependenceTy)             \
-  if (isa<SrcOpTy>(srcOp.getOperation()) &&                          \
-      isa<DstOpTy>(dstOp.getOperation()) &&                          \
-      dependenceGraph.hasDependenceFrom(srcOp, dstOp, DependenceTy)) \
-    return true;
-
-    ADD_FUSABLE_PAIR(linalg::FillOp, linalg::ConvOp,
-                     linalg::LinalgDependenceGraph::DependenceType::WAW)
-    ADD_FUSABLE_PAIR(linalg::FillOp, linalg::MatmulOp,
-                     linalg::LinalgDependenceGraph::DependenceType::WAW)
-    ADD_FUSABLE_PAIR(linalg::FillOp, linalg::PoolingMaxOp,
-                     linalg::LinalgDependenceGraph::DependenceType::WAW)
-    ADD_FUSABLE_PAIR(linalg::FillOp, linalg::PoolingMinOp,
-                     linalg::LinalgDependenceGraph::DependenceType::WAW)
-    ADD_FUSABLE_PAIR(linalg::FillOp, linalg::PoolingSumOp,
-                     linalg::LinalgDependenceGraph::DependenceType::WAW)
-
-#undef ADD_FUSABLE_PAIR
-  }
-  return false;
+  if (!dstOp || !srcOp) return true;
+
+  Optional<SmallVector<int64_t, 4>> srcLoopBounds =
+      linalg::getStaticLoopRanges(srcOp);
+  Optional<SmallVector<int64_t, 4>> dstLoopBounds =
+      linalg::getStaticLoopRanges(dstOp);
+  if (!srcLoopBounds || !dstLoopBounds) return false;
+  for (unsigned i : llvm::seq<unsigned>(
```

Comment:
Nit: I prefer to avoid having a `for` loop spanning so many lines. What about using a local variable for the min part?